integration/regression_tests.sh

```diff
@@ -0,0 +1,211 @@
+                      Default: "intermediate".
+-v    DATA_VERSION    Version of test reference data to use. 
+                      Options: "0".
+                      Default: latest release version.
+EOF
+}
+
+
+
+TEST_DATA_PARENT_DIR=/g/data/vk83/testing/um2nc/integration-tests
+
+# Default values, overwritten by command line arguments if present:
+TEST_DATA_CHOICE_DEFAULT=intermediate
+TEST_DATA_VERSION_DEFAULT=0
+CLEAN_OUTPUT=true
+
+while getopts ":-:d:hkv:" opt; do
+    case ${opt} in
+        -)
+            case ${OPTARG} in
+                help)
+                    usage
+                    exit 0
+                ;;
+                keep)
+                    CLEAN_OUTPUT=false
+                ;;
+                *)
+                    echo "Invalid option: \"--${OPTARG}\"." >&2
+                    usage
+                    exit 1
+                ;;
+            esac
+        ;;
+        d)
+            case ${OPTARG} in
+                full|intermediate|light)
+                    TEST_DATA_CHOICE=${OPTARG}
+                    ;;
+                *)
+                    echo "\"-${opt} ${OPTARG}\" is not a valid test data option. Choose between \"full\", \"intermediate\" and \"light\"." >&2
+                    usage
+                    exit 1
+                ;;
+            esac
+        ;;
+        h)
+            usage
+            exit 0
+        ;;
+        k)
+            CLEAN_OUTPUT=false
+        ;;
+        v)
+            DATA_VERSION=${OPTARG}
+        ;;
+        :)
+            echo "Option \"-${OPTARG}\" requires an argument." >&2
+            usage
+            exit 1
+        ;;
+        \?)
+            echo "Invalid option: \"-${OPTARG}\"." >&2
+            usage
+            exit 1
+        ;;
+    esac
+done
+
+# Check that no additional arguments were passed.
+if [[ -n "${@:$OPTIND:1}" ]]; then
+    echo "Invalid positional argument: \"${@:$OPTIND:1}\"." >&2
+    exit 1
+fi
+
+# Apply default data choice, version, and output directory if not set.
+echo "Using \"${TEST_DATA_CHOICE:=$TEST_DATA_CHOICE_DEFAULT}\" data."
+
+echo "Comparing to version \"${TEST_DATA_VERSION:=$TEST_DATA_VERSION_DEFAULT}\"."
+
+TEST_DATA_REFERENCE_DIR=${TEST_DATA_PARENT_DIR}/${TEST_DATA_VERSION}/${TEST_DATA_CHOICE}
+
+if [ ! -d "${TEST_DATA_REFERENCE_DIR}" ]; then
+    echo "ERROR: Test reference data directory \"${TEST_DATA_REFERENCE_DIR}\" does not exist." >&2
+    exit 1
+fi
+
+TEST_DATA_INPUT_DIR=${TEST_DATA_PARENT_DIR}/input-data
+
+if [ ! -d "${TEST_DATA_INPUT_DIR}" ]; then
+    echo "ERROR: Test input data directory \"${TEST_DATA_INPUT_DIR}\" does not exist." >&2
+    exit 1
+fi
+
+OUTPUT_DIR=$(mktemp -d)
+
+functrap() {
+    code="$?"
+    if ([ "$code" -eq 0 ] && $CLEAN_OUTPUT) || [ "$code" -eq 2 ]; then
+        rm -rf "$OUTPUT_DIR"
+    fi
+}
+trap "exit 2" SIGHUP SIGINT SIGQUIT SIGILL SIGABRT SIGTERM
+trap functrap EXIT
+
+
+echo "Binary equivalence/backwards compatibility test for um2nc."
+
```

Comment:
```suggestion
echo "Using \"${TEST_DATA_CHOICE:=$TEST_DATA_CHOICE_DEFAULT}\" data."
echo "Comparing to version \"${TEST_DATA_VERSION:=$TEST_DATA_VERSION_DEFAULT}\"."
```

Reply:
Good idea, I think this is more logical an order. I've moved the printing in e06c7d6, but kept the assignment of the `TEST_DATA_CHOICE` and `TEST_DATA_VERSION` variables at the same place, to make sure they are defined before the directory existence checks.